dask/tests/test_cli.py

```diff
@@ -34,6 +35,35 @@ def test_config_get_bad_value():
     assert result.output.startswith("Section not found")
 
 
+@pytest.fixture
+def tmp_conf_dir(tmpdir):
+    original = dask.config.PATH
+    dask.config.PATH = str(tmpdir)
+    try:
+        yield tmpdir
+    finally:
+        dask.config.PATH = original
+
+
+def test_config_set_value(tmp_conf_dir):
+    config_file = pathlib.Path(tmp_conf_dir) / "dask.yaml"
+    assert not config_file.exists()
+
+    runner = CliRunner()
+    result = runner.invoke(
+        dask.cli.config_set,
+        ["array.chunk-size", "333MiB"],
+        env={"DASK_CONFIG": str(tmp_conf_dir)},
+    )
+
+    assert config_file.exists()
+
+    expected = (
+        f"Updated [array.chunk-size] to [333MiB], config saved to {config_file}\n"
+    )
+    assert expected == result.output
```

Comment:
Could you add a test that verifies that `dask config set` does not accidentally nuke unrelated settings in the same file?